spec/datadog/opentelemetry_spec.rb

```diff
@@ -604,6 +604,51 @@
       let(:setter) { :set_attribute }
     end
 
+    describe '#record_exception' do
+      subject! do
+        start_span
+        start_span.record_exception(StandardError.new('Error'), attributes: attributes)
+        start_span.finish
+      end
+
+      let(:start_span) { otel_tracer.start_span('start-span', **span_options) }
+      let(:active_span) { Datadog::Tracing.active_span }
+
+      context 'without attributes' do
+        let(:attributes) { nil }
+        it 'sets records an event and sets the status of the span to error' do
+          expect(span.events.count).to eq(1)
+          expect(span.events[0].name).to eq('exception')
+          expect(span.events[0].time_unix_nano / 1e9).to be_within(1).of(Time.now.to_f)
+
+          expect(span.events[0].attributes.keys).to match_array(
+            ['exception.message', 'exception.type',
+             'exception.stacktrace']
+          )
+          expect(span.events[0].attributes['exception.message']).to eq('Error')
+          expect(span.events[0].attributes['exception.type']).to eq('StandardError')
+          expect(span.events[0].attributes['exception.stacktrace']).to include(":in `full_message': Error (StandardError)")
+
+          expect(span).to have_error
+        end
+      end
+
+      context 'with attributes' do
+        let(:attributes) do
+          { 'exception.stacktrace' => 'funny_stack', 'exception.type' => 'CustomError', 'exception.message' => 'NewError',
+            'candy' => true }
+        end
```

Comment:
I suggest adding test where `attributes` is an empty Hash, so we can see it falling back to the defaults.